typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -769,6 +773,19 @@
           [_ type])]))
     #f))
 
+;; True if the arities `arrs` are what we'd expect from a struct predicate
+(define (any->bool? arrs)
+  (and (= 1 (length arrs))
```

Comment:
Also the check using `string=?` doesn't look robust. We should be able to directly match on the type.